### DISCUSSION TITLE
Documentation is incredibly sparse

### DISCUSSION THREAD

**Discussion by trebon** [postmanlabs/postman-flows#402] [Defect]
Spent the past 2 hours attempting to use a request A that returns an array of objects to send requests to B.  I thought it would be simple to do the following

1. Send Request A
2. Iterate over list of objects from response
3. For each object in Response B list, send a request using Request B

i was able successfully do 1 and 2 if I send the results to a terminal using the following blocks:
Send Request -> Select -> Loop Over List -> terminal

But I cannot convince flows to treat the output of either "Loop over List" or "Iterate" (neither of which are documented.) as a variable I can reference in a Send Request block. 

Furthermore there is a "ambiguous" marker on my Select block but I can't find any documentation regarding what this means. I eventually https://github.com/postmanlabs/postman-flows/discussions/343 which is ANOTHER way to iterate over items but more importantly it shows there is some type inference happening from examples but the logic is anyone's guess.

**Comment by eugene-mednefits** [comment]
Not sure if this is accurate to your situation, but when I get "ambiguous" from Send Request data, it's because it doesn't play well with my examples response. One of the problems I found was I can't have a field to be either `String` or `null`. So I fixed it to be `String` only. But also, I have complex flows that just work even though they are "ambiguous" even though I know it'll work and the blocks are wrong.

**Comment by Ariemeth** [comment]
Have you created examples for the requests?  Flows uses the request examples to determine what the data looks like for parsing the response from a request.

**Reply by trebon** [reply]
I created an example which suddenly allowed to actually access the data in the stream but it was treating my lost of objects as a large tuple because...well I don't know there is no documentation on the flows type system. I saw in #343 that the size of the list matters. So I made my example list smaller and now i had a working data structure that wasn't "ambiguous".

**Comment by davidklemme** [comment]
Second that OP. I'll be going back and just quickly write this in javascript.
